### PR TITLE
Added Proper Docker Compose and .env.docker-setup files

### PR DESCRIPTION
**Pull request by phil-flip**
# Description

For a while now @svpernova09 and I have been planning on rewriting the [Docker Docs](https://snipe-it.readme.io/docs/docker) so it uses Docker compose, instead of Docker run. That requires us to change some stuff in the repo. Like having a  `docker-compose.yml` and `.env.docker` file which can be used by the user.

This Pr should help with getting less support inquiries in the Discord Docker channel and remove `docker run` from production deployments.

Thankies to @svpernova09 for helping me brainstorm about possible solutions and [providing the health checks](https://github.com/svpernova09/snipe-it-kickstart/blob/0595ad89275f7ed89a8dd248c9cdb1455e887598/docker/docker-compose.yml#L35) for the `docker-compose.yml`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

The updated article can be read in [the docs](https://snipe-it.readme.io/suggested-edits/6643d725ca9d4e003dac4ffa/preview).
But I tested my setup, by running the following commands.
- Clone Repo
- Rename `.env.docker` to `.env`
- Change port in `.env`, if needed
- Run `docker compose up -d` and wait for startup
- Add Content
- Run `docker compose down` to remove deployment, but not the volumes
- Run `docker compose up -d` and wait for startup
- Check if content is still there
- Run `docker compose down -v` delete deployment and data

# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation ([Not yet merged](https://snipe-it.readme.io/suggested-edits/6643d725ca9d4e003dac4ffa), waiting for this PR to be merged first)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by uberbrady — COMMENTED** [review]
This looks *AWESOME*. Since we're so close to v7, I think I'm going to look at merging this once we do the v7->develop merge. If you want to get in front of that, you can definitely re-target to the `snipeit_v7_laravel10` branch. But if you don't, I'm going to take it anyway - because this looks great and what I was hoping for. Thank you!

**Comment by phil-flip** [comment]
Thankies for the compliments. ^^
I had a feeling that because of the upcoming release there are not going to be made a lot of changes to v6. Tho I would prefer to merge it now as it should make a difference, unless the application stack changes (different DB, other backend, etc...)

File: .env.docker [hunk 1]
```diff
@@ -91,7 +96,7 @@ API_TOKEN_EXPIRATION_YEARS=40
 # --------------------------------------------
 # OPTIONAL: SECURITY HEADER SETTINGS
 # --------------------------------------------
-APP_TRUSTED_PROXIES=192.168.1.1,10.0.0.1
+APP_TRUSTED_PROXIES=192.168.1.1,10.0.0.1,172.0.0.0/8
```
Comment (jerm) [inline .env.docker:100]: This should be 172.16.0.0/12.... not all 172 addresses are private 😉

Reference: https://www.rfc-editor.org/rfc/rfc1918#section-3

**Review by jerm — COMMENTED** [review]
This looks great, but for the CIDR comment i made.

File: .env.docker (hunk 1 above)
Comment (jerm) [reply .env.docker:100]: Should we also change the 192.168 and 10 blocks to full CIDR ranges as well?

Comment (phil-flip) [reply .env.docker:100]: > This should be 172.16.0.0/12.... not all 172 addresses are private 😉
> 
> Reference: https://www.rfc-editor.org/rfc/rfc1918#section-3

Docker uses a /8 Network tho.

Comment (phil-flip) [reply .env.docker:100]: > Should we also change the 192.168 and 10 blocks to full CIDR ranges as well?

To this day I still don't know if Docker uses 172/8 or 192.168/12 at work we had issues with the second range colliding with existing networks, which was strange to see, cause to that point I always thought they used the first one (172/8)

Comment (phil-flip) [reply .env.docker:100]: I rather wish to have the underlying hostname issue fixed, rather opening up that big of a range. But I am unsure if that is possible.

**Comment by snipe** [comment]
Thank you ❤️

**Comment by phil-flip** [comment]
@snipe Thank you for merging. Mind also merging the changes in the Docs?

**Comment by snipe** [comment]
Done :)

**Comment by phil-flip** [comment]
@snipe Sorry for poking again, but this was accidentally merged into v6.3.4, instead of v7(.0.1)

**Comment by snipe** [comment]
@phil-flip Sorry, I think that's a Readme.io quirk :( Can you just give me the deets here and I'll update manually?

**Comment by phil-flip** [comment]
@snipe Well, I rewrote the entire article. You should be able to find it [here](https://snipe-it.readme.io/suggested-edits/6643d725ca9d4e003dac4ffa/preview)

**Pull request merged**